### PR TITLE
Bluetooth: controller: adv privacy fixes

### PR DESCRIPTION
**Pull request by carlescufi**

### CODE REVIEW

File: subsys/bluetooth/controller/ll_sw/ll_filter.c [hunk 1]
```diff
@@ -882,9 +886,12 @@ u32_t ll_priv_mode_set(bt_addr_le_t *id_addr, u8_t mode)
 		default:
 			return BT_HCI_ERR_INVALID_PARAM;
 		}
+	} else {
+
```
Comment (jhedberg) [inline subsys/bluetooth/controller/ll_sw/ll_filter.c]: Minor thing: redundant empty line

File: subsys/bluetooth/controller/ll_sw/ll_filter.c [hunk 2]
```diff
@@ -624,10 +624,14 @@ void ll_rl_rpa_update(bool timeout)
 		if ((rl[i].taken) && (all || !rl[i].rpas_ready)) {
 
 			if (rl[i].pirk) {
-				err = bt_rpa_create(peer_irks[rl[i].pirk_idx],
-						    &rl[i].peer_rpa);
+				u8_t irk[16];
+
+				sys_memcpy_swap(irk, peer_irks[rl[i].pirk_idx],
+						16);
+				err = bt_rpa_create(irk, &rl[i].peer_rpa);
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ll_filter.c]: Please have a bt_rpa_create_be instead that takes peer_rpa in big-endian. Saves on 2 times as h/w takes BE.

**Review by cvinayak — CHANGES_REQUESTED** [review]

**Comment by carlescufi** [comment]
Updated the PR to introduce a new big-endian RPA generation call

File: subsys/bluetooth/controller/ll_sw/ll_filter.c (hunk 1 above)
Comment (carlescufi) [inline subsys/bluetooth/controller/ll_sw/ll_filter.c]: @jhedberg addressed, can you re-review?

File: subsys/bluetooth/controller/ll_sw/ll_filter.c (hunk 2 above)
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ll_filter.c]: ok... as only Key needs to be converted to LE, current code is reasonable.

**Review by cvinayak — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Pull request merged**